pkg/descheduler/descheduler.go

```diff
@@ -153,6 +153,61 @@ func metricsProviderListToMap(providersList []api.MetricsProvider) map[api.Metri
 	return providersMap
 }
 
+// preserveNeeded returns the obj preserving fields needed for memory efficiency.
+// Only keep scheduler related fields
+func preserveNeeded(obj interface{}) (interface{}, error) {
+	// metadata related
+	if accessor, err := meta.Accessor(obj); err == nil {
+		accessor.SetManagedFields(nil)
+		accessor.SetFinalizers(nil)
+	}
+
+	if pod, ok := obj.(*v1.Pod); ok {
+		preserveContainer := func(c *v1.Container) {
+			c.Command = nil
+			c.Args = nil
+			c.WorkingDir = ""
+			c.Ports = nil
```

Comment:
I suggest not removing this c.Ports. The NodePorts plugin in the scheduler needs to use the container’s ports, and the related field values may be required during the node fit phase.